sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -396,6 +396,12 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val WHOLESTAGE_MAX_NUM_FIELDS = SQLConfigBuilder("spark.sql.codegen.maxFields")
+    .internal()
+    .doc("The maxinum number of fields (nested) could be supported by whole stage codegen")
```

Comment:
Nit: The maximum number of fields (including nested fields) that will be supported before deactivating whole-stage codegen.